packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -64,547 +43,759 @@ type TabsOrientationProps =
+  const { className, children, ...rest } = props;
+  const { hasSublabel, setHasSublabel } = useSafeContext(TabsContext);
+
+  useIsomorphicLayoutEffect(() => {
+    if (!hasSublabel) {
+      setHasSublabel(true);
+    }
+  }, [hasSublabel, setHasSublabel]);
 
   return (
     <Box
-      className={cx('iui-tabs-wrapper', `iui-${orientation}`, wrapperClassName)}
-      style={stripeProperties}
+      as='span'
+      className={cx('iui-tab-description', className)}
+      ref={ref}
+      {...rest}
     >
-      <Box
-        as='ul'
-        className={cx(
-          'iui-tabs',
-          `iui-${type}`,
-          {
-            'iui-green': color === 'green',
-            'iui-animated': type !== 'default' && isClient,
-            'iui-not-animated': type !== 'default' && !isClient,
-            'iui-large': hasSublabel,
-          },
-          tabsClassName,
-        )}
-        data-iui-overflow={overflowOptions?.useOverflow}
-        data-iui-scroll-placement={scrollingPlacement}
-        role='tablist'
-        ref={refs}
-        onKeyDown={onKeyDown}
-        {...rest}
-      >
-        {labels.map((label, index) => createTab(label, index))}
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'span'>;
+TabDescription.displayName = 'Tabs.TabDescription';
+
+// ----------------------------------------------------------------------------
+// Tabs.Actions component
+
+type TabsActionsOwnProps = {
+  /**
+   * Passes props to the wrapper component for the actions
+   */
+  wrapperProps?: React.ComponentPropsWithRef<'div'>;
+};
+
+const TabsActions = React.forwardRef((props, ref) => {
+  const { wrapperProps, className, children, ...rest } = props;
+
+  return (
+    <Box
+      {...wrapperProps}
+      className={cx('iui-tabs-actions-wrapper', wrapperProps?.className)}
+    >
+      <Box className={cx('iui-tabs-actions', className)} ref={ref} {...rest}>
+        {children}
       </Box>
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TabsActionsOwnProps>;
+TabsActions.displayName = 'Tabs.Actions';
 
-      {actions && (
-        <Box className='iui-tabs-actions-wrapper'>
-          <Box className='iui-tabs-actions'>{actions}</Box>
-        </Box>
-      )}
+// ----------------------------------------------------------------------------
+// Tabs.Panel component
+
+type TabsPanelOwnProps = {
+  /**
+   * Value used to associate the panel with a given tab.
+   */
+  value: string;
+  /**
+   * @deprecated Don't pass `id`, as it will be automatically set.
+   */
+  id?: string;
+};
+
+const TabsPanel = React.forwardRef((props, ref) => {
+  const { value, className, children, ...rest } = props;
+
+  const { activeValue, idPrefix } = useSafeContext(TabsContext);
+
+  return (
+    <Box
+      className={cx('iui-tabs-content', className)}
+      aria-labelledby={`${idPrefix}-tab-${value}`}
+      role='tabpanel'
+      hidden={activeValue !== value ? true : undefined}
+      id={`${idPrefix}-panel-${value}`}
```

Comment:
lets move this `id` after `rest` in case the user accidentally tries to override it.

same comment in Tab too

Reply:
Moved ids after rest